components/vfs/src/root_file_system.cpp

```diff
@@ -223,25 +243,23 @@ StackErrorPtr RootFileSystem::enumerate_files_recursive(const std::string &file_
 			return res;
 		}
 
-		all_files.reserve(folder_files.size());
-		for (auto &file : folder_files)
-		{
-			all_files.push_back(folder + file);
-		}
+		all_files.reserve(all_files.size() + folder_files.size());
+		all_files.insert(all_files.end(), folder_files.begin(), folder_files.end());
 	}
 
-	*files = all_files;
+	*files = std::move(all_files);
 
 	return nullptr;
 }
 
-StackErrorPtr RootFileSystem::enumerate_folders_recursive(const std::string &file_path, std::vector<std::string> *folders)
+StackErrorPtr RootFileSystem::enumerate_folders_recursive(const std::string &folder_path, std::vector<std::string> *folders)
 {
+	assert(folders);
+
 	std::set<std::string> all_dirs;
```

Comment:
As you don't need to find in `all_dirs` anymore, there's no need to use a `std::set` here, you can just use a `std::vector`.